lidarrmetadata/provider.py

```diff
@@ -714,9 +839,39 @@ def __init__(self,
         self._db_user = db_user
         self._db_password = db_password
         
+        ## dummy value for initialization, will be picked up from redis later on
+        self._last_cache_invalidation = datetime.datetime.now(pytz.utc) - datetime.timedelta(hours = 2)
+        
     def data_vintage(self):
         return self.query_from_file('../sql/data_vintage.sql')[0]['vintage']
+    
+    def invalidate_cache(self, prefix):
+
+        last_invalidation_key = prefix + 'MBProviderLastCacheInvalidation'
+        self._last_cache_invalidation = util.CACHE.get(last_invalidation_key) or self._last_cache_invalidation
```

Comment:
Could some of the repeated cache invalidation logic be improved by having a member variable and do a key based on the class name? May also be easier to just leave it to do later.